aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -628,6 +630,41 @@ public class AwsProperties implements Serializable {
    */
   public static final String LAKE_FORMATION_DB_NAME = "lakeformation.db-name";
 
+  /** Region to be used by the SigV4 protocol for signing requests. */
+  public static final String REST_SIGNER_REGION = "rest.signing-region";
+
+  /** The service name to be used by the SigV4 protocol for signing requests. */
+  public static final String REST_SIGNING_NAME = "rest.signing-name";
+
+  /** The default service name (API Gateway and lambda) used during SigV4 signing. */
+  public static final String REST_SIGNING_NAME_DEFAULT = "execute-api";
```

Comment:
nit: regarding Ryan's comment in https://github.com/apache/iceberg/pull/6951#discussion_r1125129463, there are some other variations of this internally, but publicly this is the only possible value. We could name this `REST_SIGNING_NAME_APIGATEWAY` and then do `REST_SIGNING_NAME_DEFAULT = REST_SIGNING_NAME_APIGATEWAY`, but that seems like an overkill for now we can keep it as is until any further variation come.